dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/webview/WebEventConsumerTest.kt

```diff
@@ -71,36 +71,79 @@ internal class WebEventConsumerTest {
         // Given
         val fakeBundledEvent = forge.getForgery<JsonObject>()
-        var fakeEventType = forge.anAlphabeticalString()
-        while (fakeEventType == WebEventConsumer.LOG_EVENT_TYPE) {
-            fakeEventType = forge.anAlphabeticalString()
-        }
-        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, fakeEventType)
+        val fakeRumEventType = forge.anElementFrom(WebRumEventConsumer.rumEventTypes)
+        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, fakeRumEventType)
 
         // When
         testedWebEventConsumer.consume(fakeWebEvent.toString())
 
         // Then
         verify(mockRumEventConsumer).consume(
             fakeBundledEvent,
-            fakeEventType
+            fakeRumEventType
         )
     }
 
     @Test
     fun `M delegate to LogsEventConsumer W consume() { LOG eventType }`(forge: Forge) {
         // Given
         val fakeBundledEvent = forge.getForgery<JsonObject>()
-        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, WebEventConsumer.LOG_EVENT_TYPE)
+        val fakeLogEventType = forge.anElementFrom(WebLogEventConsumer.logEventTypes)
+        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, fakeLogEventType)
+
+        // When
+        testedWebEventConsumer.consume(fakeWebEvent.toString())
+
+        // Then
+        verify(mockLogsEventConsumer).consume(fakeBundledEvent, fakeLogEventType)
+    }
+
+    @Test
+    fun `M do nothing W consume() { unknown event type }`(forge: Forge) {
+        // Given
+        val fakeBundledEvent = forge.getForgery<JsonObject>()
+        var fakeUnknownEventType = forge.anAlphabeticalString()
+        while (fakeUnknownEventType in WebRumEventConsumer.rumEventTypes ||
+            fakeUnknownEventType in WebLogEventConsumer.logEventTypes
+        ) {
+            fakeUnknownEventType = forge.anAlphabeticalString()
+        }
+        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, fakeUnknownEventType)
+
+        // When
+        testedWebEventConsumer.consume(fakeWebEvent.toString())
+
+        // Then
+        verifyZeroInteractions(mockLogsEventConsumer)
+        verifyZeroInteractions(mockRumEventConsumer)
+    }
+
+    @Test
+    fun `M log internal error event W consume() { unknown event type }`(forge: Forge) {
+        // Given
+        val fakeBundledEvent = forge.getForgery<JsonObject>()
+        var fakeUnknownEventType = forge.anAlphabeticalString()
+        while (fakeUnknownEventType in WebRumEventConsumer.rumEventTypes ||
+            fakeUnknownEventType in WebLogEventConsumer.logEventTypes
+        ) {
+            fakeUnknownEventType = forge.anAlphabeticalString()
+        }
+        val fakeWebEvent = bundleWebEvent(fakeBundledEvent, fakeUnknownEventType)
 
         // When
         testedWebEventConsumer.consume(fakeWebEvent.toString())
 
         // Then
-        verify(mockLogsEventConsumer).consume(fakeBundledEvent)
+        verify(mockSdkLogHandler).handleLog(
```

Comment:
maybe it makes sense to merge this test with the test above? they have the same Given and When sections.

Reply:
Yeah ...but one is testing the log and the other is testing that no message is sent. Following our current guidelines we kind of tend to keep these separated.